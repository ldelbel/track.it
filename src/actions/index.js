const changePage = page => ({
  type: 'change_page',
  payload: page,
});

export default changePage;