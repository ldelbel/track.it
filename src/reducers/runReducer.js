import { ADD_RUNNING_SESSION, SET_USER, FILL_LIST } from '../actions';

let initialState = {
  id: 0,
  username: '',
  runningSessions: [],
};

if (localStorage.user) {
  initialState = {
    id: localStorage.id,
    username: localStorage.user,
    runningSessions: JSON.parse(localStorage.runningSessions),
  };
}

const runReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_RUNNING_SESSION:
      return {
        ...state,
        runningSessions: state.runningSessions.concat(action.payload),
      };
    case SET_USER:
      return {
        ...state,
        username: action.payload,
      };
    case FILL_LIST:
      return {
        ...state,
        id: action.payload[0],
        runningSessions: action.payload[1],
      };
    default:
      return state;
  }
};

export default runReducer;
