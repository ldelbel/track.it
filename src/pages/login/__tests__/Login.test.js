import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Login from '../index';
import runReducer from '../../../reducers/runReducer';
import { setUser } from '../../../actions';

describe('login page', () => {
  it('renders correctly', () => {
    const store = createStore(runReducer);
    const login = renderer.create(
      <Provider store={store}>
        <Login setUser={setUser} />
      </Provider>,
    ).toJSON();
    expect(login).toMatchSnapshot();
  });
});
