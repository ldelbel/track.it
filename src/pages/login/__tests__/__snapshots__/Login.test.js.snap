// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page renders correctly 1`] = `
<main
  className="container"
>
  <div
    className="content"
  >
    <h1
      className="content__title"
    >
      Track.it
    </h1>
    <div
      className="content__input"
    >
      <span>
        Enter your name
      </span>
      <input
        data-testid="input"
        onChange={[Function]}
        type="text"
      />
      <p
        className="hidden"
      >
        Please provide a name to login
      </p>
      <input
        data-testid="login"
        onClick={[Function]}
        type="submit"
        value="Login"
      />
    </div>
  </div>
</main>
`;
